### PR TITLE
fix(intl): emit localeChanged event *after* locale has changed

### PR DESCRIPTION
**Pull request by buschtoens**
Split from https://github.com/ember-intl/ember-intl/pull/649

### CODE REVIEW

File: tests/unit/services/intl-test.js [hunk 1]
```diff
@@ -238,4 +240,16 @@ module('service:intl', function(hooks) {
...
     assert.equal(this.intl.exists('bar'), false);
   });
+
+  test('changing the locale emits the `localeChanged` event and the new locale is available', async function(assert) {
+    assert.expect(1);
+    
+    const newLocale = ['de', 'en-us'];
+
+    addListener(this.intl, 'localeChanged', () => {
+      assert.deepEqual(get(this.intl, 'locale'), newLocale);
```
Comment (jasonmit) [inline tests/unit/services/intl-test.js]: @buschtoens this gets called twice where I'd expect it to be called once.  Could be an artifact of setting the locale during the test setup, didn't get a chance to pull down the branch and investigate.

File: addon/services/intl.js [hunk 2]
```diff
@@ -38,7 +39,7 @@ export default Service.extend(Evented, {
 
       if (!isArrayEqual(proposed, this._locale)) {
         this._locale = proposed;
-        this.trigger('localeChanged');
+        next(() => this.trigger('localeChanged'));
```
Comment (jasonmit) [inline addon/services/intl.js]: Should we cancel on destroy?

Comment (jasonmit) [inline addon/services/intl.js]: Can you help me understand this issue of needing to schedule on the next runloop.  `localeChanged` is relied on in the helpers to recompute on locale change.  Is that behavior broken?

https://github.com/ember-intl/ember-intl/blob/89009718cb36ebfce0b2783e9aff05702d9f6431/addon/helpers/-format-base.js#L22

**Comment by buschtoens** [comment]
@jasonmit Thanks a lot for taking this over! ❤️

**Pull request merged**